pkg/applicationserver/linking.go

```diff
@@ -265,7 +269,12 @@ func (as *ApplicationServer) getLink(ctx context.Context, ids ttnpb.ApplicationI
 	val, ok := as.links.Load(uid)
 	if !ok {
 		if val, ok := as.linkErrors.Load(uid); ok {
-			if err := val.(error); !errors.IsCanceled(err) {
+			err := val.(error)
+			switch err.Error() {
+			case io.ErrLinkDeleted.Error():
+			case io.ErrLinkReset.Error():
```

Comment:
Use error codes checking here

It's also not safe to use our errors for comparison; in fact it's better to not expose them at all